addons/mail/models/mail_thread.py

```diff
@@ -4507,9 +4507,9 @@ def _message_update_content(self, message, body, attachment_ids=None, partner_id
         if strict:
             self._check_can_update_message_content(message.sudo())
 
-        msg_values = {
-            'body': escape(body),  # keep html if already Markup, otherwise escape
-        } if body is not None else {}
+        msg_values = {}
+        # keep html if already Markup, otherwise escape
+        msg_values["body"] = escape(body) + Markup("<span class='o-mail-Message-edited'/>") if body else False
```

Comment:
See line below `if "body" in msg_values:`
For discarding translations, it shouldn't be done if `body` is None (no change of content).

We need to make a distinction between `None` which is no change, and empty body.

Reply:
You might want to just adapt the line below to be `body is not None` instead?